presto-native-execution/presto_cpp/main/operators/ShuffleInterface.h

```diff
@@ -35,6 +35,10 @@ class ShuffleInterface {
   /// Read the next block of data for this partition.
   /// @param success set to false indicate aborted client.
   virtual velox::BufferPtr next(int32_t partition, bool success) = 0;
+  /// Return true if all the data is finished writing and is ready to
+  /// to be read while noMoreData signals the shuffle service that there
+  /// is no more data to be writen.
+  virtual bool readyForRead() const = 0;
```

Comment:
Is this method specific to testing shuffle? How would one implement it for production version of the shuffle?

Reply:
As you mentioned this method is not necessary for the batch based implementation as the reader fragments are created after all the producer fragments. But this method has two major roles:
1) Important for testing to verify the correct state of shuffle making sure the writes are complete
2) Using this we can even use the shuffle in Presto/Prestissimo and reader threads can rely on this to wait until the shuffle data is ready (again for testing before Sapphire/Velox infra is fully in place).